wfl/Profiles/FoodOrderProfile.cs

```diff
@@ -60,6 +44,51 @@ protected override void Configure(EntityConfiguration<FoodOrder> configuration)
                             o => rc => rc.GetByIdAsync(o.RestaurantId)
                         )
                     )
+                )
+                .When(
```

Comment:
Right now there are checks in the controller methods to verify that the action attempted is allowed. I could see the benefit of moving the validation checks into their own class, which would allow us to provide more specific error messages and keep the validation step in each method from looking as cluttered. Do you think that making a change like that should be treated as a separate issue / PR?